bundler/lib/dependabot/bundler/package/package_details_fetcher.rb

```diff
@@ -127,25 +115,57 @@ def fetch
         # ]
         sig { returns(Dependabot::Package::PackageDetails) }
         def rubygems_versions
-          response = registry_json_response_for_dependency
+          registry_url = get_url_from_dependency(dependency) || "https://rubygems.org"
+
+          # TODO: Github private registry support
+          # registry_url = "https://rubygems.pkg.github.com/#{OWNER_NAME}"
+          # Corresponding API URL:
+          # curl  -H "Accept: application/json" \
+          #       -H "Authorization: Bearer <<TOKEN>>" \
+          #       https://api.github.com/orgs/dsp-testing/packages/rubygems/json/version
+          begin
+            parsed_url = URI.parse(registry_url)
+            return package_details([]) if parsed_url.host == "rubygems.pkg.github.com"
+          rescue URI::InvalidURIError
+            raise "Invalid registry URL: #{registry_url}"
+          end
```

Comment:
The `begin/rescue` block should only cover what is absolutely necessary.
```suggestion
          parsed_url = begin
            URI.parse(registry_url)
          rescue URI::InvalidURIError
            raise "Invalid registry URL: #{registry_url}"
          end
          return package_details([]) if parsed_url.host == "rubygems.pkg.github.com"
```